packages/alfa-rules/src/sia-r2/rule.ts

```diff
@@ -37,22 +34,22 @@ export const SIA_R2: Atomic.Rule<Device | Document, Element> = {
       applicability: () => {
         return map(
           filter(
-            querySelectorAll(document, document, isElement, {
+            querySelectorAll(document, document, isElement(), {
               flattened: true
             }),
-            element => {
-              return map(isImage(element, document, device), isImage => {
-                if (!isImage) {
-                  return false;
-                }
-
-                if (element.localName === "img") {
-                  return true;
-                }
-
-                return isExposed(element, document, device);
-              });
-            }
+            isElement(builder =>
+              builder
+                .withNamespace(document, Namespace.HTML)
+                .browserSpecific()
+                .and(
+                  isElement(builder =>
+                    builder
+                      .withRole(device, document, Roles.Img)
+                      .and(element => isExposed(element, document, device))
+                      .or(isElement(builder => builder.withName("img")))
```

Comment:
This is a bit ugly. Especially the implicit left-associativity of the boolean operations…